internal/catalogmetadata/cache/cache.go

```diff
@@ -104,13 +114,16 @@ func (fsc *filesystemCache) writeFS(catalogName string, source io.Reader) (fs.FS
 		return nil, err
 	}
 
-	if err := os.RemoveAll(cacheDir); err != nil {
-		return nil, fmt.Errorf("error removing old cache directory: %v", err)
+	if _, err := os.Stat(cacheDir); err == nil {
```

Comment:
This seems unnecessary since `os.RemoveAll` already tolerates the case that the cache dir doesn't exist.

Reply:
It seems that os.RemoveAll costs more than  os.Stat(. 
That was the motivation.

Reply:
Is there a benchmark that can show the difference? Even if os.RemoveAll is significantly slower:
1. How often will we be running this function?
2. When this function runs, how often will cacheDir not exist?

Reply:
You are totally right, it will depends on how much times we call writeFS.
I have nothing of that for now, I am just looking to see how we can improve the cache. 
It is just an experimental staff.